lib/types/object/index.js

```diff
@@ -118,16 +118,31 @@ internals.Object = class extends Any {
                 }
             }
 
-            if (target[rename.from] === undefined) {
+
+
+            if (target[rename.from] === undefined && !(rename.from instanceof RegExp)) {
                 delete target[rename.to];
             }
+            else if (rename.from instanceof RegExp) {
+                const fromKey = Object.keys(target).find((key) => {
```

Comment:
Can we loop over this with a more traditional method instead? There's a heavy cost to using the closure here in the critical path of the validation which would be elevated through not using it.

```js
const objectKeys = Object.keys(target);

for (var i = 0, il = objectKeys.length; i < il; i++) {
  ...
}
```